### PR TITLE
add e2e test for kpt live apply

### PR DESCRIPTION
**Pull request by Liujingfang1**
The added test covers the following flow:
- `kpt live apply` with a previous version of kpt
- `kpt live preview` with built version of kpt for the same package
- `kpt live apply` with built version of kpt for the same package.

/assign @mortent @seans3

### CODE REVIEW

**Comment by Liujingfang1** [comment]
This one should be merged after #1307

File: e2e/live/end-to-end-test.sh [hunk 1]
```diff
@@ -113,6 +113,16 @@ shift $((OPTIND-1))
 #  Helper functions
 ###########################################################################
 
+function downloadPreviousKpt {
```
Comment (mortent) [inline e2e/live/end-to-end-test.sh:116]: So once we release the next version of kpt that includes support for the inventory policy, would this test still work?

**Review by mortent — COMMENTED** [review]
Would this test be better in cli-utils? It doesn't seem like this is something that is actually specific to kpt and it would be easier to set up the correct scenarios using the ginkgo tests in cli-utils.

File: e2e/live/end-to-end-test.sh (hunk 1 above)
Comment (Liujingfang1) [reply e2e/live/end-to-end-test.sh:116]: We can pin the kpt version used in this test. The last released version is v0.37.1. WDYT?

Comment (mortent) [reply e2e/live/end-to-end-test.sh:116]: That would address the first part. But is there a good reason to have this test here rather than in cli-utils?

Comment (Liujingfang1) [reply e2e/live/end-to-end-test.sh:116]: The cli-utils tests the library functionality. It doesn't test a package applied by a previous version of  the cli `kpt` can be applied by the new version of the cli `kpt`. Later the cli-utils may only contain the library part. So the `kpt` repo is a good place for this test.

Comment (mortent) [reply e2e/live/end-to-end-test.sh:116]: ok, that makes sense. But I do think we should make sure we have tests in cli-utils as well to cover the situation where a set of resources was added to the cluster without the `owning-inventory` annotation and then later is updated in way where the annotation are added. We might already cover this, but we should verify.

**Review by mortent — APPROVED** [review]

File: e2e/live/end-to-end-test.sh (hunk 1 above)
Comment (Liujingfang1) [reply e2e/live/end-to-end-test.sh:116]: Just verified that there is some e2e test covered the case without the `owning-inventory` annotation. https://github.com/kubernetes-sigs/cli-utils/blob/master/test/e2e/inventory_policy_test.go#L88

Comment (Liujingfang1) [reply e2e/live/end-to-end-test.sh:116]: Changed the version to v0.37.1. PTAL

**Pull request merged**